src/db/opt.rs

```diff
@@ -1,59 +1,143 @@
 use super::*;
+use crate::opt;
 
 #[derive(Clone)]
 pub struct AgateOptions {
+    /* Required options. */
+    /// The path of the directory where key data will be stored in.
+    ///
+    /// If it doesn't exist, Agate will try to create it for you.
     pub dir: PathBuf,
+    /// The path of the directory where value data will be stored in.
+    ///
+    /// If it doesn't exist, Agate will try to create it for you.
     pub value_dir: PathBuf,
-    // TODO: docs
-    pub in_memory: bool,
+
+    /*  Usually modified options. */
+    /// When set to true, the DB would call an additional sync after writes to flush
+    /// mmap buffer over to disk to survive hard reboots.
+    ///
+    /// The default value of `sync_writes` is false.
```

Comment:
The default value is defined below, seem no need to write again here

Reply:
There're many fields, so it's not convenient to look for default value for a certain field. Besides, if we add comment, we could easily get the value via language server such as rust-analyzer.